src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java

```diff
@@ -363,6 +376,62 @@ public Pose2d getVirtualTarget() {
             .inverse());
   }
 
+  /**
+   * Gets the pose at some time in the future
+   *
+   * @param time time in seconds
+   * @return The future pose
+   */
+  public Pose2d getFuturePose(double time) {
+    return getPose()
+        .transformBy(
+            new Transform2d(
+                getRobotRelativeSpeeds().vxMetersPerSecond * time,
+                getRobotRelativeSpeeds().vyMetersPerSecond * time,
+                new Rotation2d()));
```

Comment:
include the robot's rotational velocity (omegaRadiansPerSecond). even though we dont use it in shooting while moving its more consistent with what the method says it does